src/main.js

```diff
@@ -1,144 +1,175 @@
+// //
+const makeCompositeDisposable = () => {
+  const disposables = new Set()
+  return {
+    add: (disposable) => {
+      if (!(disposable.dispose instanceof Function)) {
+        throw new Error('disposable must have dispose method')
+      }
+      disposables.add(disposable)
+    },
+    dispose: () => disposables.forEach(d => d.dispose())
   }
 }
 
+
 module.exports = {
   activate() {
-    getMigrations().map(makeIdleCallback)
-    this.subscriptions = new CompositeDisposable()
+    // const { CompositeDisposable } = require('atom')
+    // this.subscriptions = new CompositeDisposable()
+    this.subscriptions = makeCompositeDisposable()
+
+    // Put dependency loading on Atom's todo-list
+    this.subscriptions.add(makeIdleRunner()(dependencies))
+
+    const {
+      atomConfig,
+      getMigrations,
+      subscribe: configSubscribe
+    } = require('./atom-config')
+
+    // Subscribe to Atom configuration settings
     this.subscriptions.add(...(configSubscribe()))
 
+    // Subscribe to Save event so we can run fixOnSave
     this.subscriptions.add(atom.workspace.observeTextEditors((editor) => {
       editor.onDidSave(async () => {
+        const { hasValidScope } = require('./validate/editor')
+
         const { fixOnSave, scopes } = atomConfig
         if (hasValidScope(editor, scopes) && fixOnSave) {
           await this.fixJob(true)
         }
       })
     }))
 
+    // Subscribe to Debug command
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
       'linter-eslint:debug': async () => {
-        loadDeps()
-        const debugString = await debug.report()
-        const notificationOptions = { detail: debugString, dismissable: true }
+        const { report } = require('./debug')
+
+        const debugReport = await report()
+        const notificationOptions = { detail: debugReport, dismissable: true }
         atom.notifications.addInfo('linter-eslint debugging information', notificationOptions)
       }
     }))
 
+    // Subscribe to Fix File command
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
       'linter-eslint:fix-file': async () => {
         await this.fixJob()
       }
     }))
 
+    // Add context menu entry for Fix File
     this.subscriptions.add(atom.contextMenu.add({
       'atom-text-editor:not(.mini), .overlayer': [{
         label: 'ESLint Fix',
         command: 'linter-eslint:fix-file',
         shouldDisplay: (evt) => {
           const activeEditor = atom.workspace.getActiveTextEditor()
-          if (!activeEditor) {
-            return false
-          }
-          // Black magic!
+          if (!activeEditor) return false
+
+          // Some scary voodoo black magic! Atom v1.19.0+
           // Compares the private component property of the active TextEditor
-          //   against the components of the elements
+          //   against the components property of the TextEditor DOM elements
```

Comment:
This actually makes the comment wrong, there are _many_ DOM elements that are being compared against, only one of which is likely to be a `<TextEditor>`.

Reply:
Ok. How can we make this accurate without just reverting? I think not mentioning the DOM is a pretty big commission in this comment.

Reply:
"elements" -> "DOM elements"?

Reply:
> "elements" -> "DOM elements"?
This makes sense, except unfortunately "elements" is used much more widely than just the DOM. I do not think we need to mention the dom in the code, but if we are going to take the time to write comments about it, that's a pretty bit point bring up. It's basically the only place in the code we touch the dom, so it's pretty significant that we are suddenly interacting with it. (It's only 3 letters long. 😄 ).